### PR TITLE
Apm new development

### PR DESCRIPTION
**Pull request by jedan2506**
### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update

### Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [ ] I have read the contributing guide and my code conforms to the guidelines.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have added the required tests.

### CODE REVIEW

File: superagi/controllers/analytics.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    except:
+        raise HTTPException(status_code=500, detail="Internal Server Error")
+
+
+@router.get("/agents/{agent_id}", status_code=200)
+def get_agent_runs(agent_id: int, Authorize: AuthJWT = Depends(check_auth)):
+    try:
+        return AnalyticsHelper(session=db.session).fetch_agent_runs(agent_id)
+    except:
+        raise HTTPException(status_code=500, detail=str(e))
```
Comment (Fluder-Paradyne) [inline superagi/controllers/analytics.py:37]: where is e comming from ?

File: migrations/versions/e39295ec089c_creating_events.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Create Date: 2023-06-30 12:23:12.269999
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = 'e39295ec089c'
+down_revision = '83424de1347e'
```
Comment (Fluder-Paradyne) [inline migrations/versions/e39295ec089c_creating_events.py:14]: change it to the last added migration

File: superagi/controllers/agent.py [hunk 3]
```diff
@@ -198,12 +202,18 @@ def create_agent_with_config(agent_with_config: AgentWithConfig,
     db.session.flush()
     AgentExecutionConfiguration.add_or_update_agent_execution_config(session=db.session, execution=execution,
                                                                      agent_execution_configs=agent_execution_configs)
+
+    AnalyticsHelper(session=db.session).create_event('run_created', 0, {'run_id': execution.id,'name':execution.name}, agent_id, 0),
```
Comment (TransformerOptimus) [inline superagi/controllers/agent.py:206]: dont use temp variables agent_id and agent_name

File: superagi/helper/analytics_helper.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        self.session = session
+
+    def _run_query(self, query: Query) -> Iterator:
+        try:
+            return iter(query.yield_per(100).all())
+        except SQLAlchemyError as e:
+            logging.error(f"Database error: {str(e)}")
+            return iter([])
+
+    def create_event(self, event_name: str, event_value: int, json_property: dict, agent_id: int, org_id: int) -> Optional[Event]:
```
Comment (TransformerOptimus) [inline superagi/helper/analytics_helper.py:22]: make the params optional if not mandatory.

File: superagi/controllers/agent.py (hunk 3 above)
Comment (TransformerOptimus) [inline superagi/controllers/agent.py:206]: fix naming conventions run_id, name

File: superagi/helper/analytics_helper.py (hunk 4 above)
Comment (TransformerOptimus) [inline superagi/helper/analytics_helper.py:22]: make agent_id, org_id, json_property as optional params

File: gui/pages/Content/APM/ApmDashboard.js [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+            const data = response.data;
+            setSelectedAgentRun(data)
+            averageAgentRunTime(data)
+        })
+            .catch((error) => {
+                console.error(`There was a problem with the request: ${error}`);
+            });
+    }
+
+    const averageAgentRunTime = (runs) => {
```
Comment (TransformerOptimus) [inline gui/pages/Content/APM/ApmDashboard.js:153]: you should run aggregated query in backend in backend for this.

File: superagi/helper/analytics_helper.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+                "agent_id": agent_id,
+                "runs_completed": 0,
+                "total_calls": 0,
+                "total_tokens": 0
+            }
+
+            model = event.json_property.get('model', 'Unknown')
+            models_used_dict[model] = models_used_dict.get(model, 0) + 1
+
+        for event in run_completed_events:
```
Comment (TransformerOptimus) [inline superagi/helper/analytics_helper.py:63]: you should aggregate in db and get this query.

File: superagi/helper/analytics_helper.py [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+            result['runs_completed'] += 1
+
+        return result
+
+    def fetch_agent_data(self) -> Dict[str, List[Dict[str, int]]]:
+        agent_created_events = self._run_query(self.session.query(Event).filter_by(event_name="agent_created"))
+        run_completed_events = self._run_query(self.session.query(Event).filter_by(event_name="run_completed"))
+
+        agent_details_dict, models_used_dict = {}, {}
+        for event in agent_created_events:
```
Comment (TransformerOptimus) [inline superagi/helper/analytics_helper.py:50]: you cant run a loop like this. It should be aggregated query.

File: superagi/helper/analytics_helper.py [hunk 8]
```diff
@@ -0,0 +1,202 @@
+from typing import List, Dict, Tuple, Optional, Iterator
+from sqlalchemy.orm import Session
+from sqlalchemy.orm.query import Query
+from superagi.models.events import Event
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy import text, func
+import logging
+
+class AnalyticsHelper:
```
Comment (TransformerOptimus) [inline superagi/helper/analytics_helper.py:9]: move this to analytics folder instead of helper

**Pull request closed without merging**